frontend/src/pages/Home.tsx

```diff
@@ -249,6 +250,41 @@ export default function Home() {
           <MovingLogos sponsors={data.sponsors} />
         </SecondaryCard>
       </div>
+      <div className="mb-20 mt-8">
+        <SecondaryCard title="Upcoming Events">
+          <div className="grid gap-4 sm:grid-cols-1 md:grid-cols-2 lg:grid-cols-3">
+            {data.upcomingEvents.map((event: EventType) => (
+              <div key={event.name} className="rounded-lg bg-gray-200 p-4 dark:bg-gray-700">
+                <h3 className="mb-2 text-lg font-semibold">
+                  <a
+                    href={event.url}
+                    className="hover:underline"
+                    target="_blank"
+                    rel="noopener noreferrer"
+                  >
+                    {event.name}
+                  </a>
+                </h3>
+                <div className="mb-2">
+                  <p>{event.description}</p>
+                </div>
+                <div className="flex flex-wrap items-center text-sm text-gray-600 dark:text-gray-300">
+                  <div className="mr-4 flex items-center">
```

Comment:
Let's keep it simple and instead Mar 13, 2025 Mar 13, 2025 show Mar 13 , 2025 if start/end date is the same or range if they differ. Let's minimize redundancy.